docs/src/architecture/assets.md

```diff
@@ -0,0 +1,61 @@
+
+## Native assets
+Native assets are data structures issued by asset accounts and can be stored in account and note vaults. All native assets are encoded using a single `Word` (4 field elements), so it is easy to determine the type of an asset inside and outside Miden VM. The asset encodes both the ID of the issuing account and asset details. So, when we look at an asset, we know who issued it. It is guaranteed that there can never be a collision between a fungible and a non-fungible asset.
+
+### Issuance
+Specialized asset accounts called faucets issue assets by creating notes. Like any other account, faucets have an Account ID called `faucet_id`. Users can define specific properties of the asset when creating the faucet. Asset creation is similar to creating ERC20 contracts on Ethereum. The rules for who can issue assets and how are fully customizable and defined by the code deployed in the faucet account. The only restrictions are:
+
+* max supply for fungible assets, and 
+* uniqueness of non-fungible assets.
+
+Faucets can create assets and immediately distribute them by producing notes. However, assets can also stay in the faucet after creation to be sent later, e.g., in a bundle. That way, one can mint a million NFTs locally in a single transaction and then send them out as needed in separate transactions in the future.
+
+<p align="center">
+    <img src="../diagrams/architecture/asset/Asset_Issuance.png">
+</p>
+
+### Fungible assets
+A fungible asset is encoded using the amount and the `faucet_id` of the faucet which issued the asset. The amount is guaranteed to be `2^63 - 1` or smaller, the maximum supply for any fungible asset. Examples of fungible assets are ETH and stablecoins, e.g., DAI, USDT, and USDC.
+
+If the `faucet_id` of MATIC were to be `2`, 100 MATIC are encoded as `[100, 0, 0, 2]` - whereas the `0`s in the middle help to quickly distinguish between fungible and non-fungible assets. 
+
+Note: It is impossible to find a collision between two fungible assets issued by different faucets as the `faucet_id` is included in the asset's description. The id is guaranteed to be other for each faucet as per the faucet creation logic.
+
+### Non-fungible assets
+A non-fungible asset is encoded by hashing the asset data into a `Word` and the second element with the `faucet_id` that issued the asset. It looks like `[e0, faucet_id, e2, e3]`. Note that the second element is guaranteed to be non-Zero.
+
+Examples of non-fungible assets are all NFTs, e.g., a DevCon ticket. Here the asset data can be the owner and the ticket number.
+
+It is impossible to find a collision between two non-fungible assets issued by different faucets as the `faucet_id` is included in the description of the non-fungible asset, which is guaranteed to be different per the faucet creation logic. Collision resistance for non-fungible assets issued by the same faucet is `~2^95`.
```

Comment:
We can actually guarantee that there are no collisions between assets issues by the same faucet - so, the statement about $2^{95}$ can be deleted.

In general, I'd either remove this from here, or maybe have a separate small section which talks about collisions between assets.

Reply:
Got it, thanks.